src/Markup/Avalonia.Markup.Xaml/SourceInfo/SourceInfo.cs

```diff
@@ -0,0 +1,215 @@
+﻿using System;
+using Avalonia.Controls;
+
+namespace Avalonia.Markup.Xaml.SourceInfo
+{
+    /// <summary>
+    /// Represents source location information for an element within a XAML or code file.
+    /// </summary>
+    /// <remarks>
+    /// This struct is typically used to store the line, column, and file path of a control or object
+    /// to enable navigation or diagnostics (for example, jumping to a definition in Visual Studio).
+    /// </remarks>
+    public readonly struct SourceInfo : IEquatable<SourceInfo>
```

Comment:
It's not a good idea to [name a class the same as its containing namepace](https://learn.microsoft.com/en-us/archive/blogs/ericlippert/do-not-name-a-class-the-same-as-its-namespace-part-one)